src/main/java/com/selfxdsd/todos/PuzzlesComponent.java

```diff
@@ -57,153 +50,68 @@
-                if(issue.body().contains(puzzle.getId())) {
-                    foundIssue = true;
-                    break;
-                }
-            }
-            if(!foundIssue) {
-                final Issue newIssue = issues.open(
-                    puzzle.issueTitle(),
-                    puzzle.issueBody(),
-                    Puzzle.PUZZLE_LABEL,
-                    String.format(
-                        Puzzle.ESTIMATION_LABEL,
-                        puzzle.getEstimate()
-                    )
-                );
-                opened.add("#" + newIssue.issueId());
-            }
-        }
-        if(opened.size() > 0) {
-            String author = commit.author();
-            if(author != null && !author.isEmpty()) {
-                author = "@" + author + " ";
-            } else {
-                author = "";
-            }
-            commit.comments().post(
-                author + "I've opened the Issues "
-                + opened + " for the newly added to-dos.\n\n"
-                + "The to-dos may have been added in an earlier commit, "
-                + "but I've found them just now."
-            );
-        }
+        this.beanFactory = beanFactory;
+        this.reviewer = reviewer;
     }
 
     /**
-     * Close issues which don't have a corresponding puzzle
-     * (puzzle has been removed from code).
-     * @param puzzles Puzzles found in the repo.
-     * @param issues Issues API.
-     * @param commit Commit which triggered everything.
+     * Listener for {@link PuzzlesWebhookEvent} Spring event triggered
+     * on webhook request.
+     * @param event Puzzles Webhook Event.
+     **@throws Exception If something goes wrong on closing database connection.
      */
-    private void closeRemovedPuzzles(
-        final Puzzles<Project> puzzles,
-        final Issues issues,
-        final Commit commit
-    ) {
-        final List<String> closed = new ArrayList<>();
-        for(final Issue issue : issues) {
-            boolean foundPuzzle = false;
-            for(final Puzzle puzzle : puzzles) {
-                if(issue.body().contains(puzzle.getId())) {
-                    foundPuzzle = true;
-                    break;
-                }
-            }
-            if(!foundPuzzle && !issue.isClosed()) {
-                issue.close();
-                issue.comments().post(
-                    "Puzzle disappeared from the code, "
-                    + "that's why I closed this ticket."
-                );
-                closed.add("#" + issue.issueId());
-            }
-        }
-        if(closed.size() > 0) {
-            String author = commit.author();
-            if(author != null && !author.isEmpty()) {
-                author = "@" + author + " ";
+    @Async
+    @EventListener
+    public void onPuzzleWebhookEvent(final PuzzlesWebhookEvent event)
+        throws Exception {
+        final Self selfCore = this.beanFactory.getBean(Self.class);
```

Comment:
fresh connection here in 'async', notice that `PuzzlesWebhookEvent ` is a simple data class, with no reference to a Project object, and thus not exposing other db connection (`Storage`).